src/core/macros.systemd.in

```diff
@@ -130,16 +140,20 @@ SYSTEMD_INLINE_EOF \
 #   %tmpfiles_create_package %{name} %SOURCE1
 #   %files
 #   %{_tmpfilesdir}/%{name}.conf
+
 %tmpfiles_create_package() \
-systemd-tmpfiles --replace=%_tmpfilesdir/%1.conf --create - <<SYSTEMD_INLINE_EOF >/dev/null 2>&1 || : \
+%{expand:%%{?!__systemd_twoargs_%#:%%{error:This macro requires two arguments}}} \
+[ -x @bindir@/systemd-tmpfiles ] && @bindir@/systemd-tmpfiles --replace=%_tmpfilesdir/%1.conf --create - <<SYSTEMD_INLINE_EOF || : \
 %(cat %2) \
 SYSTEMD_INLINE_EOF \
 %{nil}
 
 %sysctl_apply() \
-@rootlibexecdir@/systemd-sysctl %{?*} >/dev/null 2>&1 || : \
+%{expand:%%{?__systemd_someargs_%#}} \
+[ -x @rootlibexecdir@/systemd-sysctl ] && @rootlibexecdir@/systemd-sysctl %{?*} || : \
 %{nil}
 
 %binfmt_apply() \
-@rootlibexecdir@/systemd-binfmt %{?*} >/dev/null 2>&1 || : \
+%{expand:%%{?__systemd_someargs_%#}} \
+[ -x @rootlibexecdir@/systemd-binfmt ] && @rootlibexecdir@/systemd-binfmt %{?*} || : \
```

Comment:
The argument checking part is pure genius, I love it.

There seems to have been a misunderstanding. Removing the error redirection for systemd-tmpfiles, systemd-sysctl, systemd-binfmt, systemd-sysusers is not OK, please add that part back.

Reply:
Can you give a reason why it is supposedly not ok? If there is a syntax error in (e.g.) /usr/lib/tmpfiles.d/wine.conf, I would want to see it, just like the sysusers/tmpfiles case.

```
# echo ":" >broken.conf
# /usr/lib/systemd/systemd-binfmt wilinglybroken.conf
Rule file name '' is not valid, refusing.
Failed to add binary format: Invalid argument
```

Reply:
Please see https://github.com/systemd/systemd/pull/12048#discussion_r267280686. rpm scriptlets are not the right place to see errors in configuration of other packages.

Reply:
But it is generally not going to show errors from other packages, because the 0-argument invocation has been ruled out, and n-argument invocation only processes the files so given, and those files are from this package itself.

Reply:
No, when `--replace=` is used, the full configuration that is a merge of the configuration on disk and the stuff specified inline is executed. So it's entirely possible to have errors from files provided by other packages.